booking.py

```diff
@@ -206,6 +207,35 @@ def add_tagged_friends(friends):
                         f2.tag = t
         return friends
 
+    def check_validity(root_file_path, validation_dict, friends):
+        root_file = TFile(root_file_path)
+        quantities = set([x.GetName() for x in root_file.Get("ntuple").GetListOfLeaves()])
+        friend_quantitites = set()
+        for f in friends:
+            friend = TFile(f)
+            friend_quantitites.update(set([x.GetName() for x in friend.Get("ntuple").GetListOfLeaves()]))
+        # first we check the main ntuple, then the friends
+        errordata = {}
+        if len(validation_dict["varset"]) == 0:
+            validation_dict["varset"] = quantities
+        else:
+            difference = (validation_dict["varset"] - quantities) | (quantities - validation_dict["varset"])
```

Comment:
Why not use symmetric difference here?
```suggestion
            difference = validation_dict["varset"] ^ quantities
```

Reply:
i did not know about this one ! Changed